src/PrpShop/PRP/Render/QPlasmaCubicRender.h

```diff
@@ -0,0 +1,76 @@
+/* This file is part of PlasmaShop.
+ *
+ * PlasmaShop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PlasmaShop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PlasmaShop.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _QPLASMACUBICRENDER_H
+#define _QPLASMACUBICRENDER_H
+
+#include <QOpenGLBuffer>
+#include <QOpenGLWidget>
+#include <QOpenGLTexture>
+#include <QOpenGLFunctions>
+#include <QOpenGLShaderProgram>
+
+#include <PRP/Surface/plCubicEnvironmap.h>
+
+class plMipmap;
+class QPushButton;
+class QPlasmaCubicRender : public QOpenGLWidget, protected QOpenGLFunctions {
+    Q_OBJECT
+
+private:
+    QOpenGLShaderProgram fGLProgram;        // Shader program for rendering the cube.
+    QOpenGLBuffer        fGLArrayBuf;       // Buffer for the cube vertex positions and texture coordinates.
+    QOpenGLBuffer        fGLIndexBuf;       // Buffer for the cube geometry.
+    QOpenGLTexture      *fGLTexture;        // GL texture of fCubeMap.
+    QOpenGLTexture      *fGLLabelTexture;   // GL texture of fFaceLabelTexture.
+    QQuaternion          fGLRotation;       // Rotation of the cube/viewport in OpenGL space.
+    QMatrix4x4           fGLProjection;     // Projection matrix for the OpenGL viewport.
+
+    QImage        fCubeMap;              // A texture atlas containing the six faces of the environment map.
+    QImage        fFaceLabelTexture;     // A texture atlas containing the face labels to be used as an overlay.
+    QPoint        fMouseFrom;            // Mouse position when the button was pressed.
+    QPushButton  *fShowFaceLabelsOption; // Button to toggle the visibility of face labels.    
```

Comment:
These would be nice to have as doxygen style

Reply:
FYI, these display correctly in Intellisense tooltips as-is.